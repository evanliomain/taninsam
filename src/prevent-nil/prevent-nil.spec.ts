import { preventNil } from './prevent-nil';

describe('preventNil function', () => {
  const input = [1, 2, 3, 4, 5];
  let f;

  beforeEach(() => {
    f = jest.fn().mockReturnValue(1);
  });

  test('is a pure function', () => {
    preventNil(f)(input);
    expect(input).toMatchSnapshot();
  });

  test('undefined |> preventNil(f) === 1', () => {
    expect(preventNil(f)(undefined)).toBe(undefined);
    expect(f).not.toHaveBeenCalled();
  });

  test('null |> preventNil(f) === 1', () => {
    expect(preventNil(f)(null)).toBe(undefined);
    expect(f).not.toHaveBeenCalled();
  });

  test('1 |> preventNil(f) === 1', () => {
    expect(preventNil(f)(1)).toBe(1);
    expect(f).toHaveBeenCalledWith(1);
  });
});
