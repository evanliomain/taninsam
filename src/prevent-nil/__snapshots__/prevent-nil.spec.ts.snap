// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preventNil function is a pure function 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;
